HW2/P3/mandelbrot.pyx

```diff
@@ -31,16 +70,74 @@ cpdef mandelbrot(np.complex64_t [:, :] in_coords,
     assert in_coords.shape[0] == out_counts.shape[0], "Input and output arrays must be the same size"
     assert in_coords.shape[1] == out_counts.shape[1],  "Input and output arrays must be the same size"
 
+    # convert to complex part and real part
+    in_reals = np.real(in_coords)
+    in_imags = np.imag(in_coords)
+    
+    num_threads = 4
+     
+    # for reseting purposes
+    zeros8 = AVX.float_to_float8(0.0)
+
+    # used in comparison later
+    fours8 = AVX.float_to_float8(4.0)
+    nans8 = AVX.greater_than(fours8,zeros8)
+   
+    
     with nogil:
-        for i in range(in_coords.shape[0]):
-            for j in range(in_coords.shape[1]):
-                c = in_coords[i, j]
-                z = 0
+        for i in prange(in_coords.shape[0],num_threads=num_threads,schedule='static',chunksize=1):
+            for j_idx in range(in_coords.shape[1]/8): # grab sets of 8 columns
+                j = j_idx * 8 # the real offset
+                
+                # get list of the complex values in the 8 columns
+                # and convert to float8
+                # there must be a better way to do this
+                reals8 = AVX.make_float8(in_reals[i,j+7], in_reals[i,j+6], in_reals[i,j+5], in_reals[i,j+4], in_reals[i,j+3], in_reals[i,j+2], in_reals[i,j+1], in_reals[i,j])
+                imags8 = AVX.make_float8(in_imags[i,j+7], in_imags[i,j+6], in_imags[i,j+5], in_imags[i,j+4], in_imags[i,j+3], in_imags[i,j+2], in_imags[i,j+1], in_imags[i,j])
+                
+                #reset float8 vectors for z's
+                z_reals8 = AVX.bitwise_and(zeros8, zeros8)
```

Comment:
This is an interesting approach. Why not just use "AVX.float_to_float8(0.0)" for both of these? Is there some perf gain you get out of using the AVX bitwise and?